src/internal/connector/onedrive/drive.go

```diff
@@ -89,30 +88,31 @@ func drives(
 
 	// Loop through all pages returned by Graph API.
 	for {
+		var (
+			err  error
+			page gapi.PageLinker
+		)
+
 		// Retry Loop for Drive retrieval. Request can timeout
 		for i := 0; i <= numberOfRetries; i++ {
 			page, err = pager.GetPage(ctx)
 			if err != nil {
 				// Various error handling. May return an error or perform a retry.
-				detailedError := support.ConnectorStackErrorTraceWrap(err, "").Error()
-				if strings.Contains(detailedError, userMysiteURLNotFound) ||
-					strings.Contains(detailedError, userMysiteURLNotFoundMsg) ||
-					strings.Contains(detailedError, userMysiteNotFound) ||
-					strings.Contains(detailedError, userMysiteNotFoundMsg) {
+				errMsg := support.ConnectorStackErrorTraceWrap(err, "").Error()
+				if strings.Contains(errMsg, userMysiteURLNotFound) ||
+					strings.Contains(errMsg, userMysiteURLNotFoundMsg) ||
+					strings.Contains(errMsg, userMysiteNotFound) ||
+					strings.Contains(errMsg, userMysiteNotFoundMsg) {
```

Comment:
Probably should make this a function

Reply:
actually, I did!  It's in one of the coming PRs.